test/modules/TwinTester/Program.cs

```diff
@@ -22,20 +22,20 @@ static async Task Main()
 
             try
             {
-                await Task.Delay(Settings.Current.TestStartDelay);
-
-                RegistryManager registryManager = RegistryManager.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString);
-
-                using (ITwinTestInitializer twinOperator = await GetTwinOperatorAsync(registryManager))
+                using (RegistryManager registryManager = RegistryManager.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString))
                 {
-                    await twinOperator.Start();
+                    ITwinTestInitializer twinOperator = await GetTwinOperatorAsync(registryManager);
+                    await twinOperator.StartAsync(cts.Token);
                     await Task.Delay(Settings.Current.TestDuration, cts.Token);
-                }
 
-                await cts.Token.WhenCanceled();
-                completed.Set();
-                handler.ForEach(h => GC.KeepAlive(h));
-                Logger.LogInformation("TwinTester exiting.");
+                    Logger.LogInformation($"Test run completed after {Settings.Current.TestDuration}");
+                    twinOperator.Stop();
```

Comment:
do you need to call stop explicitly when exception caught?

Reply:
when there is an exception here the program ends.